dotCMS/src/test/java/com/dotmarketing/db/DataSourceStrategyProviderTest.java

```diff
@@ -0,0 +1,128 @@
+            throws IllegalAccessException, InstantiationException, ClassNotFoundException {
+
+        final DataSourceStrategyProvider provider = Mockito.spy(DataSourceStrategyProvider.class);
+        final SystemEnvironmentProperties properties = Mockito.mock(SystemEnvironmentProperties.class);
+        final DBPropertiesDataSourceStrategy dbStrategy = Mockito.mock(DBPropertiesDataSourceStrategy.class);
+        final SystemEnvDataSourceStrategy systemEnvStrategy = Mockito.mock(SystemEnvDataSourceStrategy.class);
+        final DockerSecretDataSourceStrategy dockerSecretStrategy = Mockito.mock(DockerSecretDataSourceStrategy.class);
+        final TomcatDataSourceStrategy tomcatDataSourceStrategy = Mockito.mock(TomcatDataSourceStrategy.class);
+
+        final HikariDataSource dummyDatasource = new HikariDataSource();
+
+        Mockito.when(provider.getSystemEnvironmentProperties()).thenReturn(properties);
+        Mockito.when(provider.getDBPropertiesInstance()).thenReturn(dbStrategy);
+        Mockito.when(provider.getSystemEnvDataSourceInstance()).thenReturn(systemEnvStrategy);
+        Mockito.when(provider.getDockerSecretDataSourceInstance()).thenReturn(dockerSecretStrategy);
+        Mockito.when(provider.getTomcatDataSourceInstance()).thenReturn(tomcatDataSourceStrategy);
+
+        if (testCase.equals("DBProperties")) {
+            Mockito.when(dbStrategy.existsDBPropertiesFile()).thenReturn(true);
+        } else {
+            Mockito.when(dbStrategy.existsDBPropertiesFile()).thenReturn(false);
+        }
+
+        if (testCase.equals("SystemEnv")){
+            Mockito.when(properties.getVariable("connection_db_base_url")).thenReturn("dummy_url");
+        } else {
+            Mockito.when(properties.getVariable("connection_db_base_url")).thenReturn(null);
+        }
+
+        if (testCase.equals("DockerSecret")) {
+            Mockito.when(dockerSecretStrategy.dockerSecretPathExists()).thenReturn(true);
+        } else {
+            Mockito.when(dockerSecretStrategy.dockerSecretPathExists()).thenReturn(false);
+        }
+
+        Mockito.when(dbStrategy.apply()).thenReturn(dummyDatasource);
+        Mockito.when(systemEnvStrategy.apply()).thenReturn(dummyDatasource);
+        Mockito.when(dockerSecretStrategy.apply()).thenReturn(dummyDatasource);
+        Mockito.when(tomcatDataSourceStrategy.apply()).thenReturn(dummyDatasource);
+
+
+        //Gets the provider strategy
+        DataSource result = provider.get();
+
+        assertNotNull(result);
+        assertEquals(dummyDatasource, result);
+
+        Mockito.verify(dbStrategy, Mockito.times(testCase.equals("DBProperties")? 1: 0)).apply();
+        Mockito.verify(systemEnvStrategy, Mockito.times(testCase.equals("SystemEnv")? 1: 0)).apply();
+        Mockito.verify(dockerSecretStrategy, Mockito.times(testCase.equals("DockerSecret")? 1: 0)).apply();
+        Mockito.verify(tomcatDataSourceStrategy, Mockito.times(testCase.equals("Tomcat")? 1: 0)).apply();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5026767)